src/xbot/common/controls/sensors/adapters/InertialMeasurementUnitAdapter.java

```diff
@@ -0,0 +1,74 @@
+package xbot.common.controls.sensors.adapters;
+
+import edu.wpi.first.wpilibj.I2C;
+import edu.wpi.first.wpilibj.SPI;
+import edu.wpi.first.wpilibj.SerialPort;
+import edu.wpi.first.wpilibj.I2C.Port;
+import xbot.common.controls.sensors.NavImu;
+import xbot.common.controls.sensors.XGyro;
+import xbot.common.controls.sensors.navx.AHRS;
+import xbot.common.math.ContiguousDouble;
+
+public class InertialMeasurementUnitAdapter extends NavImu implements XGyro {
+
+    AHRS ahrs;
+    boolean isBroken = false;
+
+    public InertialMeasurementUnitAdapter() {
+        super(ImuType.navX);
+        /* Options: Port.kMXP, SPI.kMXP, I2C.kMXP or SerialPort.kUSB */
+        try {
+            this.ahrs = new AHRS(Port.kMXP);
+        }
+        catch (Exception e){
+            isBroken = true;
+        }
+        this.ahrs.zeroYaw();
```

Comment:
Hmm... good point. There are some edge cases where it might be a good idea for the IMU to keep state. It's hard to imagine they'd be hit, but better safe then sorry.